### PR TITLE
Api 7558 customers retention

### PR DESCRIPTION
**Pull request by czeczot**

### CODE REVIEW

File: agent/api_test.go [hunk 1]
```diff
@@ -887,6 +888,9 @@ func TestListCustomersShouldReturnDataReceivedFromAgentAPI(t *testing.T) {
 	if nextPage != "" {
 		t.Errorf("Invalid next page ID: %v", nextPage)
 	}
+	if limit != 1 {
+		t.Errorf("Invalid limit: %v", total)
```
Comment (kogierman) [inline agent/api_test.go:892]: limit -> limited customers amount

File: agent/api_test.go [hunk 2]
```diff
@@ -870,7 +871,7 @@ func TestListCustomersShouldReturnDataReceivedFromAgentAPI(t *testing.T) {
 		t.Errorf("API creation failed")
 	}
 
-	customers, total, prevPage, nextPage, rErr := api.ListCustomers(100, "page", "asc", "", agent.NewCustomersFilters())
+	customers, total, limit, prevPage, nextPage, rErr := api.ListCustomers(100, "page", "asc", "", agent.NewCustomersFilters())
```
Comment (kogierman) [inline agent/api_test.go:874]: limit -> limited

**Review by kogierman — APPROVED** [review]

**Review by kogierman — APPROVED** [review]

File: agent/api_test.go [hunk 3]
```diff
@@ -887,6 +888,9 @@ func TestListCustomersShouldReturnDataReceivedFromAgentAPI(t *testing.T) {
 	if nextPage != "" {
 		t.Errorf("Invalid next page ID: %v", nextPage)
 	}
+	if limited != 1 {
+		t.Errorf("Invalid limited customers amount: %v", total)
```
Comment (litwinow) [inline agent/api_test.go:892]:
```suggestion
		t.Errorf("Invalid limited customers amount: %v", limited)
```

File: agent/internal.go [hunk 4]
```diff
@@ -181,8 +181,9 @@ type listCustomersRequest struct {
 
 type listCustomersResponse struct {
 	hashedPaginationResponse
-	Customers      []objects.Customer `json:"customers"`
-	TotalCustomers uint               `json:"total_customers"`
+	Customers        []objects.Customer `json:"customers"`
+	TotalCustomers   uint               `json:"total_customers"`
+	LimitedCustomers uint               `json:"limited_customers,omitempty"`
```
Comment (litwinow) [inline agent/internal.go:186]: `,omitempty` not needed, it doesn't do anything on unmarshalling

**Review by litwinow — CHANGES_REQUESTED** [review]

**Review by litwinow — APPROVED** [review]

**Comment by czeczot** [comment]
already merged

**Pull request closed without merging**